libs/ktxreader/src/Ktx2Reader.cpp

```diff
@@ -137,14 +176,166 @@ void Ktx2Reader::unrequestFormat(Texture::InternalFormat format) {
 }
 
 Texture* Ktx2Reader::load(const uint8_t* data, size_t size, TransferFunction transfer) {
```

Comment:
This method seems pretty similar to `doTranscoding`, is there a way for them to share their implementation? Maybe a helper method that returns an `std::optional<Texture::PixelBufferDescriptor>`.